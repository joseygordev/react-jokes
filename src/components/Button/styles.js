import theme from '../../config/theme';

export default {
  root: {
    fontWeight: 'bold',
    fontSize: '1rem',
    margin: `${theme.spacing.margin * 1.4}px 0px`,
    textTransform: 'uppercase',
    padding: `${theme.spacing.padding}px ${theme.spacing.padding * 3}px`,
    background: theme.colors.secondary,
    borderRadius: 5,
    cursor: 'pointer',
    lineHeight: 1.4,
    color: theme.colors.white,
    width: '100%',
  },
  mobile: {
    width: 'auto',
  },
  hover: {
    boxShadow: '0 0.125rem 0.25rem rgba(0,0,0,0.25)',
    opacity: .75
  }
};
