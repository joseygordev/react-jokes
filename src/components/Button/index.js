import React, {useState} from 'react';
import PropTypes from 'prop-types';

import {metrics} from '../../config/metrics';

import {useMediaQuery} from '../../hooks/useMediaQuery';

// Styles
import styles from './styles';

const Button = ({label, onClick}) => {
  const [hoverActive, setHoverActive] = useState(false);
  const isPageWide = useMediaQuery(metrics.tablet);

  return (
    <button 
      style={{...styles.root, ...(isPageWide && styles.mobile), ...(hoverActive && styles.hover)}} 
      onMouseEnter={() => setHoverActive(true)}
      onMouseLeave={() => setHoverActive(false)} 
      type="button" 
      onClick={onClick}>
      {label}
    </button>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
Button.defaultProps = {
};

export default Button;
