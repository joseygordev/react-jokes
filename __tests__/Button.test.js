import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button';

describe('Button', () => {
  it('Should return the Button component', () => {
    const tree = renderer
      .create(<Button label="My button" onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
