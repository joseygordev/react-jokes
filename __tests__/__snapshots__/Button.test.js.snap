// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should return the Button component 1`] = `
<button
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#8a8a8a",
      "borderRadius": 5,
      "color": "#FFFFFF",
      "cursor": "pointer",
      "fontSize": "1rem",
      "fontWeight": "bold",
      "lineHeight": 1.4,
      "margin": "11.2px 0px",
      "padding": "8px 24px",
      "textTransform": "uppercase",
      "width": "100%",
    }
  }
  type="button"
>
  My button
</button>
`;
